bndtools.m2e/src/bndtools/m2e/BndConfigurator.java

```diff
@@ -145,20 +150,15 @@ public AbstractBuildParticipant getBuildParticipant(final IMavenProjectFacade pr
 					return build;
 				}
 
-				// There might be more than one bnd-maven-plugin execution on
-				// any single project. An example of this would be bundling test
-				// classes. However, each such execution would be pointing at a
-				// particular output directory, for example:
-				//
-				// ${project.build.outputDirectory}
-				// vs.
-				// ${project.build.testOutputDirectory}
-				//
-				// With that in mind the scope of detectable changes needs to be
-				// limited to the actual output directory associated with the
-				// plugin by reading the classesDir property.
-				final File classesDir = maven.getMojoParameterValue(projectFacade.getMavenProject(), execution,
-					"classesDir", File.class, monitor);
+
+				Set<IPath> deltaFilePaths = collectResourceDeltas(getDelta(project)).map(IResourceDelta::getResource)
+					.filter(IFile.class::isInstance)
+					.map(IResource::getFullPath)
+					.collect(Collectors.toSet());
+
+				final String targetDirectory = projectFacade.getMavenProject()
```

Comment:
Doesn't this lose checking executions for different target directories as the removed comment says?

Reply:
Yes. I discussed with Ray and he agreed this more "course grained" approach would be ok provided we didn't his the "infinite build" loop problem, which I have tried to address with the new resource delta check.

Reply:
@bjhargrave you might be right, but @gamerson and I discussed this and this change solves some problems that @kriegfrj reported and it's on me to test if this breaks the use case I documented in the comment.

Reply:
OK, as long as you got this! :-)